examples/evaluation/evaluate_with_static_dataset.py

```diff
@@ -0,0 +1,34 @@
+import shap
+import xgboost
+from sklearn.model_selection import train_test_split
+
+import mlflow
+
+# Load the UCI Adult Dataset
+X, y = shap.datasets.adult()
+
+# Split the data into training and test sets
+X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.33, random_state=42)
+
+# Fit an XGBoost binary classifier on the training data split
+model = xgboost.XGBClassifier().fit(X_train, y_train)
+
+# Build the Evaluation Dataset from the test set
+y_test_pred = model.predict(X=X_test)
+eval_data = X_test
+eval_data["label"] = y_test
+eval_data["prediction"] = y_test_pred
+
+
```

Comment:
@liangz1 I modified this example to try out specifying a Pandas Dataset:

```
import shap
import xgboost
from sklearn.model_selection import train_test_split

import mlflow

# Load the UCI Adult Dataset
X, y = shap.datasets.adult()

# Split the data into training and test sets
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.33, random_state=42)

# Fit an XGBoost binary classifier on the training data split
model = xgboost.XGBClassifier().fit(X_train, y_train)

# Build the Evaluation Dataset from the test set
y_test_pred = model.predict(X=X_test)
eval_data = X_test
eval_data["label"] = y_test
eval_data["prediction"] = y_test_pred

# INSERTED THIS LINE
eval_data = mlflow.data.from_pandas(eval_data)


with mlflow.start_run() as run:
    # Evaluate the static dataset without providing a model
    result = mlflow.evaluate(
        data=eval_data,
        targets="label",
        predictions="prediction",
        model_type="classifier",
        evaluators=["default"],
    )

print(f"metrics:\n{result.metrics}")
print(f"artifacts:\n{result.artifacts}")
```

I ran into this error, even though `targets` is specified:

```
Traceback (most recent call last):
  File "/Users/corey.zumar/mlflow/examples/evaluation/evaluate_with_static_dataset.py", line 27, in <module>
    result = mlflow.evaluate(
  File "/Users/corey.zumar/mlflow/mlflow/models/evaluation/base.py", line 1645, in evaluate
    raise MlflowException(
mlflow.exceptions.MlflowException: The targets argument is required when data is a Dataset and does not define targets.
```

Reply:
Just realized that we don't support passing an `mlflow.data.PandasDataset`, but I think we should. It's a bit confusing to allow pandas datafram without `PandasDataset`, especially since we're adding `predictions` to `PandasDataset` just for this use case.

Reply:
Added